packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1505,14 +1603,22 @@ export class SnapController extends BaseController<
    * @param newSnapState - The new state of the snap.
    * @param encrypted - A flag to indicate whether to use encrypted storage or not.
    */
-  updateSnapState(snapId: SnapId, newSnapState: string, encrypted: boolean) {
-    this.update((state) => {
-      if (encrypted) {
-        state.snapStates[snapId] = newSnapState;
-      } else {
-        state.unencryptedSnapStates[snapId] = newSnapState;
-      }
-    });
+  async updateSnapState(
+    snapId: SnapId,
+    newSnapState: Record<string, Json>,
+    encrypted: boolean,
+  ) {
+    if (encrypted) {
+      const encryptedState = await this.#encryptSnapState(snapId, newSnapState);
+
+      this.update((state) => {
```

Comment:
Isn't it cleaner to have a single call to update like before?

Reply:
Do you think this is more readable?

```ts
  async updateSnapState(
    snapId: SnapId,
    newSnapState: Record<string, Json>,
    encrypted: boolean,
  ) {
    const newState = encrypted
      ? await this.#encryptSnapState(snapId, newSnapState)
      : JSON.stringify(newSnapState);

    this.update((state) => {
      if (encrypted) {
        state.snapStates[snapId] = newState;
      } else {
        state.unencryptedSnapStates[snapId] = newState;
      }
    });
  }
```

Reply:
I was thinking:
```ts
  async updateSnapState(
    snapId: SnapId,
    newSnapState: Record<string, Json>,
    encrypted: boolean,
  ) {
    this.update((state) => {
      if (encrypted) {
        const encryptedState = await this.#encryptSnapState(snapId, newSnapState);
        state.snapStates[snapId] = encryptedState;
      } else {
        state.unencryptedSnapStates[snapId] = JSON.stringify(newSnapState);
      }
    });
  }
```

Reply:
I don't think the update callback can be asynchronous

Reply:
Oh I see.